src/main/java/com/telefonica/iot/cygnus/sinks/OrionSink.java

```diff
@@ -255,16 +322,70 @@ private void persist(Event event) throws Exception {
             // notification
             throw new Exception("Unrecognized content type (not Json nor XML)");
         } // if else if
+        
+        return notification;
+    } // parseEventBody
+    
+    private void accumulateInBatch(Map<String, String> headers, NotifyContextRequest notification) {
+        Long recvTimeTs = new Long(headers.get(Constants.HEADER_TIMESTAMP));
+        String service = headers.get(Constants.HEADER_NOTIFIED_SERVICE);
+        String[] defaultServicePaths = headers.get(Constants.HEADER_DEFAULT_SERVICE_PATHS).split(",");
+        String[] defaultDestinations = headers.get(Constants.HEADER_DEFAULT_DESTINATIONS).split(",");
+        
+        for (int i = 0; i < defaultDestinations.length; i++) {
+            String destination = defaultDestinations[i];
+            ArrayList<CygnusEvent> list = defaultBatch.getEvents(destination);
+            
+            if (list == null) {
+                list = new ArrayList<CygnusEvent>();
+                defaultBatch.addEvents(destination, list);
+            } // if
+            
+            CygnusEvent cygnusEvent = new CygnusEvent(
+                    recvTimeTs, service, defaultServicePaths[i], destination,
+                    notification.getContextResponses().get(i).getContextElement());
+            list.add(cygnusEvent);
+        } // for
+        
+        String[] groupedServicePaths = headers.get(Constants.HEADER_GROUPED_SERVICE_PATHS).split(",");
+        String[] groupedDestinations = headers.get(Constants.HEADER_GROUPED_DESTINATIONS).split(",");
 
-        persist(eventHeaders, notification);
-    } // persist
+        for (int i = 0; i < groupedDestinations.length; i++) {
+            String destination = defaultDestinations[i];
+            ArrayList<CygnusEvent> list = groupedBatch.getEvents(destination);
+            
+            if (list == null) {
+                list = new ArrayList<CygnusEvent>();
+                groupedBatch.addEvents(destination, list);
+            } // if
+            
+            CygnusEvent cygnusEvent = new CygnusEvent(
+                    recvTimeTs, service, groupedServicePaths[i], destination,
+                    notification.getContextResponses().get(i).getContextElement());
+            list.add(cygnusEvent);
+        } // for
+    } // accumulateInBatch
+    
+    private void doBatchRollback() {
+        // TBD: tech debt
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/b13eedc0daf7fb440f3d3decb43b596a54cb9f1f